native/cocos/audio/graph_based/AudioBuffer.cpp

```diff
@@ -0,0 +1,81 @@
+#include "audio/graph_based/AudioBuffer.h"
+#include "LabSound/extended/AudioFileReader.h"
+#include "FileUtils.h"
+namespace cc {
+ccstd::unordered_map<ccstd::string, IntrusivePtr<AudioBuffer>> AudioBuffer::bufferMap;
+AudioBuffer* AudioBuffer::createBuffer(const ccstd::string& url) {
```

Comment:
Why needs to add `createBuffer` to AudioBuffer class?
In the webaudio standard, there is not a method like that.
https://www.w3.org/TR/webaudio/#audiobuffer
```
[[Exposed](https://heycam.github.io/webidl/#Exposed)=Window]
interface AudioBuffer {
  [constructor](https://www.w3.org/TR/webaudio/#dom-audiobuffer-audiobuffer) ([AudioBufferOptions](https://www.w3.org/TR/webaudio/#dictdef-audiobufferoptions) options);
  readonly attribute [float](https://heycam.github.io/webidl/#idl-float) [sampleRate](https://www.w3.org/TR/webaudio/#dom-audiobuffer-samplerate);
  readonly attribute [unsigned long](https://heycam.github.io/webidl/#idl-unsigned-long) [length](https://www.w3.org/TR/webaudio/#dom-audiobuffer-length);
  readonly attribute [double](https://heycam.github.io/webidl/#idl-double) [duration](https://www.w3.org/TR/webaudio/#dom-audiobuffer-duration);
  readonly attribute [unsigned long](https://heycam.github.io/webidl/#idl-unsigned-long) [numberOfChannels](https://www.w3.org/TR/webaudio/#dom-audiobuffer-numberofchannels);
  [Float32Array](https://heycam.github.io/webidl/#idl-Float32Array) [getChannelData](https://www.w3.org/TR/webaudio/#dom-audiobuffer-getchanneldata) ([unsigned long](https://heycam.github.io/webidl/#idl-unsigned-long) channel);
  [undefined](https://heycam.github.io/webidl/#idl-undefined) [copyFromChannel](https://www.w3.org/TR/webaudio/#dom-audiobuffer-copyfromchannel) ([Float32Array](https://heycam.github.io/webidl/#idl-Float32Array) destination,
                             [unsigned long](https://heycam.github.io/webidl/#idl-unsigned-long) channelNumber,
                             optional [unsigned long](https://heycam.github.io/webidl/#idl-unsigned-long) bufferOffset = 0);
  [undefined](https://heycam.github.io/webidl/#idl-undefined) [copyToChannel](https://www.w3.org/TR/webaudio/#dom-audiobuffer-copytochannel) ([Float32Array](https://heycam.github.io/webidl/#idl-Float32Array) source,
                           [unsigned long](https://heycam.github.io/webidl/#idl-unsigned-long) channelNumber,
                           optional [unsigned long](https://heycam.github.io/webidl/#idl-unsigned-long) bufferOffset = 0);
};
```

Reply:
WebAudio standard uses AudioContext.decodeAudioData to get a AudioBuffer.

https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext/decodeAudioData

Reply:
decodeAudioData return a promise, so the interface is implemented in TS, createBuffer used in CPP only.